quil-rs/src/parser/lexer/mod.rs

```diff
@@ -139,15 +139,24 @@ fn _lex(input: LexInput) -> InternalLexResult<Vec<TokenWithLocation>> {
     terminated(
         many0(alt(
             "indentation or a token preceded by whitespace",
-            (
-                token_with_location(value(Token::Indentation, tag("    "))),
-                preceded(many0(tag(" ")), lex_token),
-            ),
+            (lex_indent, preceded(many0(tag(" ")), lex_token)),
```

Comment:
Thinking we should remove the `preceded(many0(tag(" "))` part, since we're being more strict about indentation being four spaces.

```suggestion
            (lex_indent, lex_token),
```

Reply:
Yeah, I was expecting to remove that too, but it becomes clear pretty quickly that `preceded` is actually necessary to catch spaces between instructions.

Reply:
Do you have a concrete example of this? Instructions should not be separated by spaces afaik

Reply:
Oh yeah, instructions was the wrong word there, I meant `tokens`. Having an issue with my environment, but I'll share an example of how removing the preceded parser causes an error.

Reply:
Oh, that makes more sense. In that case, it seems like `separated_list0` or `separated_list1` makes more sense

Reply:
Hmm, I could be missing my shot in nom golf, but I can't figure out how to make that combinator work for this case. It doesn't satisfy the trait bounds of the other wrapping combinators.

Reply:
I believe my intent was something like this, which does change the signature of the right-hand parser from `TokenWithLocation` to `Vec<TokenWithLocation>`, but also seems more correct?

```suggestion
            (lex_indent, separated_list1(space1, lex_token)),
```

Reply:
I suppose, in order to make this "more correct", newline parsing would have to be handled by `lex_indent` (since an indent is specifically a newline followed by four spaces). That seems like it would cause more code changes, since it seems like newlines are currently handled by `lex_token`?